pkg/scaffold/resource/resource.go

```diff
@@ -79,8 +79,11 @@ func (r *Resource) Validate() error {
 		return fmt.Errorf(
 			"version must match ^v\\d+(alpha\\d+|beta\\d+)?$ (was %s)", r.Version)
 	}
-	if r.Kind != flect.Pascalize(r.Kind) {
-		return fmt.Errorf("kind must be PascalCase (expected %s was %s)", flect.Pascalize(r.Kind), r.Kind)
```

Comment:
My understand to solve the https://github.com/kubernetes-sigs/kubebuilder/issues/1091 is just replace the text in the msg `CamelCase` to `PascalCase` which shows done already.

Reply:
This PR will remove `Pascalize` completely though (in favor of the kube-like validation), and the new error messages describe what went wrong (hopefully! Open to feedback on that 😄 )

Reply:
Solved can close this one. great work 👍